weaviate/embedded.py

```diff
@@ -98,11 +117,15 @@ def check_supported_platform(self):
 
     def start(self):
         if self.is_listening():
-            print(f"embedded weaviate is already listing on port {self.port}")
+            print(f"embedded weaviate is already listing on port {self.options.port}")
             return
 
         self.ensure_weaviate_binary_exists()
         my_env = os.environ.copy()
+        if self.options.additional_env_vars is not None:
+            for key, value in self.options.additional_env_vars.items():
+                my_env.setdefault(key, value)
+
```

Comment:
Use this in line 139 after all setdefault calls are done:
```suggestion
if self.options.additional_env_vars is not None:
    my_env.update(self.options.additional_env_vars)
```